### PR TITLE
mon/PGMonitor: batch filter pg states; add sanity check

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

**Comment by xiexingguo** [comment]
retest this please

**Comment by xiexingguo** [comment]
@tchaikov Some mon-related minor fixes, could you kindly help to review? Thanks:-)

File: src/mon/PGMonitor.cc [hunk 1]
```diff
@@ -621,15 +621,18 @@ void PGMonitor::handle_statfs(MonOpRequestRef op)
...
-  dout(10) << "handle_statfs " << *statfs << " from " << statfs->get_orig_source() << dendl;
 
   if (statfs->fsid != mon->monmap->fsid) {
-    dout(0) << "handle_statfs on fsid " << statfs->fsid << " != " << mon->monmap->fsid << dendl;
+    dout(0) << "handle_statfs on fsid " << statfs->fsid
+            << " != " << mon->monmap->fsid << dendl;
     return;
   }
 
+  MStatfsReply *reply;
```
Comment (tchaikov) [inline src/mon/PGMonitor.cc]: could move this line down to where `reply` is used for the first time.

File: src/mon/PGMonitor.cc [hunk 2]
```diff
@@ -935,23 +935,15 @@ void PGMonitor::check_osd_map(epoch_t epoch)
...
-    propose = true;
-  }
-
-  if (map_pg_creates())
-    propose = true;
-  if (register_new_pgs())
-    propose = true;
+  assert(pg_map.last_osdmap_epoch < epoch);
+  pending_inc.osdmap_epoch = epoch;
+  (void)map_pg_creates();
```
Comment (tchaikov) [inline src/mon/PGMonitor.cc]: `check_osd_map()` is the only caller of these functions. maybe we can change them so that they return `void` instead?

**Comment by xiexingguo** [comment]
@tchaikov Done and thanks.

File: src/mon/PGMap.cc [hunk 3]
```diff
@@ -1754,24 +1754,17 @@ void PGMap::generate_test_instances(list<PGMap*>& o)
   }
 }
 
-void PGMap::get_filtered_pg_stats(const string& state, int64_t poolid, int64_t osdid,
+void PGMap::get_filtered_pg_stats(__u32 state, int64_t poolid, int64_t osdid,
```
Comment (tchaikov) [inline src/mon/PGMap.cc]: `__u32` is a linux specific type, maybe we can try to use `uint32_t` instead? i see the underlying type of `pg_state_t::state` is `__u32`. but we can be more portable by using these ANSI types.

Comment (xiexingguo) [inline src/mon/PGMap.cc]: Done. And @tchaikov , I see you are busy with cmake, thank you for your time:-)

File: src/osd/osd_types.cc [hunk 4]
```diff
@@ -824,7 +824,7 @@ std::string pg_state_string(int state)
     oss << "repair+";
   if ((state & PG_STATE_BACKFILL_WAIT) &&
       !(state &PG_STATE_BACKFILL))
-    oss << "wait_backfill+";
+    oss << "backfill_wait+";
```
Comment (tchaikov) [inline src/osd/osd_types.cc]: could update the `doc/dev/osd_internals/recovery_reservation.rst` accordingly.

Comment (xiexingguo) [inline src/osd/osd_types.cc]: Done. Thank you for your reminder.

File: src/mon/PGMonitor.h [hunk 5]
```diff
@@ -115,17 +115,13 @@ class PGMonitor : public PaxosService {
...
-   *
-   * @return true if we updated pending_inc (and should propose)
    */
-  bool register_new_pgs();
+  void register_new_pgs();
 
   /**
    * recalculate creating pg mappings
-   *
-   * @return true if we updated pending_inc
```
Comment (tchaikov) [inline src/mon/PGMonitor.h]: could you update the comment accordingly?

Comment (xiexingguo) [inline src/mon/PGMonitor.h]: Well, I don't understand what do you mean. It does not return anything now.

Comment (tchaikov) [inline src/mon/PGMonitor.h]: ah, sorry, i thought you missed this.

**Comment by tchaikov** [comment]
lgtm

**Comment by xiexingguo** [comment]
@tchaikov @liewegas 

Tested at: http://www.daisycloud.org:9091/xxg-2016-07-02_16:59:55-rados:basic-wip-xxg-testing-2016-06-25---basic-plana/

**Comment by xiexingguo** [comment]
@tchaikov Thanks for the merge.

**Pull request merged**